src/celeritas/global/detail/AlongStepActionImpl.hh

```diff
@@ -177,7 +177,13 @@ inline CELER_FUNCTION void along_step_track(CoreTrackView const& track)
         step_limit.step = geo_step;
     }
 
-    // TODO: update track's lab-frame time here
+    // Update track's lab-frame time using the beginning-of-step velocity
+    {
+        real_type velocity = native_value_from(particle.speed());
+        CELER_ASSERT(velocity > 0);
```

Comment:
Technically since it's a scalar it should be speed 😉

Reply:
yep you got me :)